src/message/messagesActions.js

```diff
@@ -112,16 +114,21 @@ export const fetchNewer = () => (dispatch: Dispatch, getState: GetState): Action
   }
 };
 
-export const doNarrow = (newNarrow: Narrow, anchor: number = Number.MAX_SAFE_INTEGER): Action => (
+export const doNarrow = (newNarrow: Narrow, anchor?: number): Action => (
   dispatch: Dispatch,
   getState: GetState,
 ) => {
   const auth = getAuth(getState());
   const anyMessagesInNewNarrow = JSON.stringify(newNarrow) in getAllMessages(getState());
-
-  if (!anyMessagesInNewNarrow) {
+  if (!anyMessagesInNewNarrow && !anchor) {
```

Comment:
I think we are currently only passing anchor to doNarrow in case of message header clicks. If we also want to pass anchor as 0 to imply cases where we want to anchor to first unread message, then I'll also incorporate that?